components/job-orchestration/job_orchestration/scheduler/compress/compression_scheduler.py

```diff
@@ -0,0 +1,319 @@
+        except Exception as e:
+            logger.error(f"Error while getting results for job {job_id}: {e}")
+            job_success = False
+
+        db_conn.commit()
+
+        if job_success:
+            logger.info(f"Job {job_id} succeeded.")
+            update_compression_job_metadata(db_cursor, job_id, dict(
+                status=CompressionJobStatus.SUCCEEDED,
+                duration=duration,
+                uncompressed_size=uncompressed_size,
+                compressed_size=compressed_size,
+                num_tasks_completed=num_tasks_completed
+            ))
+        else:
+            logger.error(f"Job {job_id} failed. See worker logs or status_msg for details.")
+            update_compression_job_metadata(db_cursor, job_id, dict(
+                status=CompressionJobStatus.FAILED,
+                status_msg=error_message,
+                num_tasks_completed=num_tasks_completed
+            ))
+        db_conn.commit()
+
+        jobs_to_delete.append(job_id)
+
+    for job_id in jobs_to_delete:
+        del scheduled_jobs[job_id]
+
+
+def main(argv):
+    args_parser = argparse.ArgumentParser()
+    args_parser.add_argument('--config', '-c', required=True, help='CLP configuration file.')
+    args = args_parser.parse_args(argv[1:])
+
+    # Setup logging
+    log_file = Path(os.getenv("CLP_LOGS_DIR")) / "compression_scheduler.log"
+    logging_file_handler = logging.FileHandler(filename=log_file, encoding="utf-8")
+    logging_file_handler.setFormatter(get_logging_formatter())
+    logger.addHandler(logging_file_handler)
+
+    # Update logging level based on config
+    set_logging_level(logger, os.getenv("CLP_LOGGING_LEVEL"))
+
+    # Load configuration
+    config_path = Path(args.config)
+    try:
+        clp_config = CLPConfig.parse_obj(read_yaml_config_file(config_path))
+    except ValidationError as err:
+        logger.error(err)
+        return -1
+    except Exception as ex:
+        logger.error(ex)
+        # read_yaml_config_file already logs the parsing error inside
+        return -1
+
+    logger.info('Starting compression scheduler')
+    sql_adapter = SQL_Adapter(clp_config.database)
+
+    with closing(sql_adapter.create_connection(True)) as db_conn, \
+            closing(db_conn.cursor(dictionary=True)) as db_cursor:
+        # Start Job Processing Loop
+        while True:
+            try:
+                search_and_schedule_new_tasks(db_conn, db_cursor, sql_adapter.database_config.
+                                              get_clp_connection_params_and_type(True))
+                poll_running_jobs(db_conn, db_cursor)
+                time.sleep(clp_config.compression_scheduler.jobs_poll_delay)
+            except KeyboardInterrupt:
+                logger.info('Gracefully shutting down')
+                break
+            except Exception as ex:
+                logger.exception(f"Error in scheduling: {ex}")
+                break
```

Comment:
```suggestion
            except Exception:
                logger.exception(f"Error in scheduling.")
                return -1
```

`logger.exception` will automatically print the exception.